pkg/build/builder/cmd/dockercfg/cfg.go

```diff
@@ -93,6 +121,21 @@ func readDockercfg(filePath string) (cfg credentialprovider.DockerConfig, err er
 	return
 }
 
+// readDockerConfigJson reads the contents of a .docker/config.json file into a map
+// with server name keys and AuthEntry values
+func readDockerConfigJson(filePath string) (cfg credentialprovider.DockerConfig, err error) {
+	content, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return
+	}
+	var config credentialprovider.DockerConfigJson
```

Comment:
No, structs can go on the heap.

On Thu, Jan 28, 2016 at 11:12 PM, Ben Parees <notifications@github.com>
wrote:

> In pkg/build/builder/cmd/dockercfg/cfg.go
> <https://github.com/openshift/origin/pull/6887#discussion_r51224162>:
>
> > @@ -93,6 +121,21 @@ func readDockercfg(filePath string) (cfg credentialprovider.DockerConfig, err er
> >  	return
> >  }
> >
> > +// readDockerConfigJson reads the contents of a .docker/config.json file into a map
> > +// with server name keys and AuthEntry values
> > +func readDockerConfigJson(filePath string) (cfg credentialprovider.DockerConfig, err error) {
> > +	content, err := ioutil.ReadFile(filePath)
> > +	if err != nil {
> > +		return
> > +	}
> > +	var config credentialprovider.DockerConfigJson
>
> whether it goes to the stack or the heap is moot, right? structs are
> always initialized so there's no need to explicitly initialize it, end of
> story.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6887/files#r51224162>.
>